crates/storage/provider/src/either_writer.rs

```diff
@@ -16,6 +18,32 @@ pub enum EitherWriter<'a, CURSOR, N> {
     StaticFile(StaticFileProviderRWRefMut<'a, N>),
 }
 
+#[derive(Debug, EnumIs)]
+#[allow(missing_docs)]
+pub enum EitherWriterDestination {
+    Database,
+    StaticFile,
+}
+
+impl EitherWriter<'_, (), ()> {
+    /// Returns the destination for writing receipts.
```

Comment:
this needs docs about what the rules are here

Reply:
done